### ISSUE TITLE
Request: Method to Activate/Deactivate for automation

### ISSUE THREAD

**Issue by NocturneBlock** [AdAway/AdAway#995]
I use E-Robot to automate various functions on my device that require evaluative logic. There are various other similar services such as Tasker and Automate. I also use a very few apps which include ads in such a way that I am rewarded for watching them to a sufficient degree that I want ads allowed while those apps are active. As it stands, I have to manually enable and disable ads when I start and stop those apps, but if AdAway provided hooks to disable or re-enable ad blocking, I could set up E-Robot to do so, and not accidentally forget to turn my blocking back on after using those apps.

I feel this is a valuable alternative to the whitelist, as these apps frequently use popular ad providers that would pop up in considerably more apps than I would like if whitelisted, and per-app whitelisting seems to be out of scope for AdAway itself.

If it is not excessively difficult, the ability to turn the tcp dump on and off using a similar method would also be beneficial, as it would allow the user to acquire a dump from when the specific app being evaluated had focus (by creating an automation script that only activated tcpdump under that condition), making it easier to narrow down which domains are relevant to that specific app.

**Comment by AlphaDE** [comment]
I support this, I would like to ensure that (for example after OTA update with the new A/B layout), AdAway gets reenabled at the first boot.

**Comment by patrickdrd** [comment]
subscribe me

**Comment by PerfectSlayer** [comment]
Do you have any developer documentation for this king of integration?

For now, I just add an exported broadcast receiver to command adblocking status (pause and resume it). It should allow other applications to send message to control it.

Moreover, with the next version will also come the capability to exclude an app from ad blocking. It might be better in some cases rather than disabling the whole blocking.

**Comment by hcc-israel** [comment]
https://tasker.joaoapps.com/plugins-intro.html

**Comment by NocturneBlock** [comment]
Tamas Barta is the developer for E-Robot, and docs aren't really his strong suit, but if it works for Tasker, it will likely also work for E-Robot. If not, I am open to switching services anyway.

App-based whitelist also sounds amazing. If it is feasible and not too much of a processing drain, I would love to be able to have a mini-white/blacklist for a specific app. There are many apps that use one ad service for obnoxious, intrusive ads, and a different ones for rewarded ads that the user manually consents to or requests. It would be awesome if it was possible to allow just the one app to connect to the ad network it is using respectfully, while still blocking the ones it is being obnoxious wuth, and without changing the general protection.

If that is out of scope for this issue, let me know, and I will add it as a separate enhancement request. Also, whether or not that is possible, thank you for the app-level activation.

**Comment by Cris--** [comment]
I would find it also very useful to toggle AdAway completely on and off with Tasker! A simple activity for that in AdAway is enough.

**Comment by PerfectSlayer** [comment]
Finally! I spend few nights on this issue and eventually pin it!

I already developed the feature on AdAway but not fully test it and users often get back to me telling they can't make it work with Tasker. In fact, there was several issues:
- The AdAway application requires a custom permission marked as _dangerous_ so it is not granted by default.
I don't want any application be able to freely disable adblocking
- The Android system now requires to specify the (Android application) package the broadcast targets, otherwise it won't be delivered.

The second issue is easily fixed by settings the package to `org.adaway` but the first one will require Tasker developer to update its application manifest to request the AdAway custom permission so your task can request it send send broadcast intents to command AdAway... And this could take time.

In the mean time, I developed another application that will freely accept all requests from other apps and forward them to AdAway. This is for compatibility only (and also to provide a sample demo with its source code for any developer that would like to play with). You can find it [here](https://github.com/AdAway/AdAwayAutomation) (no release yet) and it will be compatible with the coming 5.8.0 version of AdAway.

Stay tuned for the next release and I will update the AdAway wiki to add automation documentation 😉

**Comment by Cris--** [comment]
Great PerfectSlayer but i need a little help.

Where must be _COMMAND_ and _STOP_ or _START_ placed on Tasker action Send Intent?

**Comment by PerfectSlayer** [comment]
Hi @Cris-- 

I just wrote the documentation: https://github.com/AdAway/AdAway/wiki/Automation
You will find capture about settings to use for `Send intent` configuration.

Also, keep in mind you will need a third app (AdAway automation) until Tasker be updated. All details in the wiki 😉

**Comment by Cris--** [comment]
Ahh, Wiki updated now.. 😁 👍
All is clear now. thx!

**Comment by guerda** [comment]
> I just wrote the documentation: https://github.com/AdAway/AdAway/wiki/Automation You will find capture about settings to use for `Send intent` configuration.
> 
> Also, keep in mind you will need a third app (AdAway automation) until Tasker be updated. All details in the wiki 😉

It looks like the wiki is outdated and the automation app is not available anymore in Google Play or F-Droid.
Is there still a possibility to control AdAway from Tasker and the sorts?

**Comment by JonyWater** [comment]
> Hi @Cris--
> 
> I just wrote the documentation: https://github.com/AdAway/AdAway/wiki/Automation You will find capture about settings to use for `Send intent` configuration.
> 
> Also, keep in mind you will need a third app (AdAway automation) until Tasker be updated. All details in the wiki 😉

Hey, the wiki is lacking the most important information:     
A link for the adaway automation app 🙃    
Only after looking in this issue's thread did I find that app    
@PerfectSlayer can you please update the wiki page so it points people in the right direction?
 
> In the mean time, I developed another application that will freely accept all requests from other apps and forward them to AdAway. This is for compatibility only (and also to provide a sample demo with its source code for any developer that would like to play with). You can find it [here](https://github.com/AdAway/AdAwayAutomation)
>

**Comment by guerda** [comment]
> A link for the adaway automation app 🙃\nOnly after looking in this issue's thread did I find that app

The problem is that I don't find the app in any app store anymore. I only find the GitHub repo

**Comment by PerfectSlayer** [comment]
As far as I remember, it's only available on the GitHub release page:
* [Release page](https://github.com/AdAway/AdAwayAutomation/releases)
* [AdAwayAutomation-1.0.0.apk](https://github.com/AdAway/AdAwayAutomation/releases/download/v1.0.0/AdAwayAutomation-1.0.0.apk)

**Comment by guerda** [comment]
Thanks. While this works from time to time, it seems that even though without any battery optimization for all three apps (AdAway, AdAway automation and Tasker), switching it off and on only works randomly.

Is there even a chance that Tasker would support the intent for AdAway? Is there a feature request for this?
